android/src/main/java/io/keen/client/android/AndroidJsonHandler.java

```diff
@@ -40,8 +44,27 @@
 
         String json = readerToString(reader);
         try {
-            JSONObject jsonObject = new JSONObject(json);
-            return JsonHelper.toMap(jsonObject);
+            Object jsonObjOrArray = getJsonObjectManager().newTokener(json).nextValue();
+
+            // TODO : What should the return value be? Technically it could be a List or Map, so it
+            // should be Object, then client code would need to do the instanceof check. For now,
+            // so as to not break the KeenJsonHandler interface, we can stick a dummy "root" key in
+            // the map we pass back.
```

Comment:
Is there anywhere in the code where this isn't passed a map? I believe our API will always respond in a map format.

Either way, I think I'm fine with creating a fake root object so it becomes a map... but I'm not terribly opinionated on the matter.